restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java

```diff
@@ -1361,8 +1362,16 @@ public void onConnectivityEvent(String jobId, RCDeviceListener.RCConnectivitySta
    {
       RCLogger.i(TAG, "onConnectivityEvent(): id: " + jobId + ", connectivityStatus: " + connectivityStatus);
       cachedConnectivityStatus = connectivityStatus;
-      if (state == DeviceState.OFFLINE && connectivityStatus != RCDeviceListener.RCConnectivityStatus.RCConnectivityStatusNone) {
-         state = DeviceState.READY;
+
+      storageManagerPreferences = new StorageManagerPreferences(this);
+      boolean pushEnabled = storageManagerPreferences.getBoolean(ParameterKeys.PUSH_NOTIFICATIONS_ENABLE_PUSH_FOR_ACCOUNT, false);
+      String binding = storageManagerPreferences.getString(FcmConfigurationHandler.FCM_BINDING, null);
+      boolean registeredForPush = binding!=null;
+
+      if (state == DeviceState.OFFLINE && connectivityStatus != RCDeviceListener.RCConnectivityStatus.RCConnectivityStatusNone){
+         if (pushEnabled && registeredForPush) {
```

Comment:
If pushEnabled is false shouldn't we set state to READY?

Reply:
You are right, I have testing though and it was working without READY state. Will fix.

Reply:
cool